# Instrucciones

1. Determine el modelo de regresión lineal que le permita entender la relación entre la variable PIB_per_cápita y las variables regresoras Mortalidad_Infantil,	Tasa_Alfabetismo_Femenino	y	Tasa_fecundidad.
2. ¿Qué puede concluir respecto de los estadísticos globales del modelo?
3. ¿Qué puede concluir respecto de los estadísticos individuales del modelo?

In [ ]:
import pandas as pd
import numpy as np
import scipy.stats as ss
import statsmodels.formula.api as smf # Nos permite ajustar el modelo de regresión

/usr/local/lib/python3.7/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [ ]:
# Importamos la base de datos y la denominamos df por "data frame"
df = pd.read_csv('solemne_forma_1.csv', sep=';', decimal=",")

In [ ]:
df

Mortalidad_Infantil  ...  Tasa_fecundidad
0                   128  ...             6.66
1                   204  ...             6.15
2                   202  ...             7.00
3                   197  ...             7.25
4                    96  ...             3.81
..                  ...  ...              ...
59                  115  ...             3.89
60                  186  ...             6.90
61                   47  ...             4.10
62                  178  ...             6.09
63                  142  ...             7.20

[64 rows x 4 columns]

In [ ]:
df.describe()

Mortalidad_Infantil  ...  Tasa_fecundidad
count            64.000000  ...        64.000000
mean            139.921875  ...         5.565469
std              77.260291  ...         1.521400
min              12.000000  ...         1.690000
25%              78.500000  ...         4.615000
50%             138.500000  ...         6.040000
75%             192.500000  ...         6.660000
max             312.000000  ...         8.490000

[8 rows x 4 columns]

# Solución

1.

In [ ]:
# Usamos el método de los mínimos cuadrados para ajustar la regresión lineal multiple
Modelo = smf.ols('PIB_per_cápita ~ Mortalidad_Infantil+	Tasa_Alfabetismo_Femenino+Tasa_fecundidad', df).fit() #Ajustamos el modelo
# Usamos summary para pedir el resultado del modelo
Modelo.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:         PIB_per_cápita   R-squared:                       0.164
Model:                            OLS   Adj. R-squared:                  0.122
Method:                 Least Squares   F-statistic:                     3.931
Date:                Thu, 18 Mar 2021   Prob (F-statistic):             0.0125
Time:                        02:25:54   Log-Likelihood:                -590.85
No. Observations:                  64   AIC:                             1190.
Df Residuals:                      60   BIC:                             1198.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
=============================================================================================
                                coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------
Intercept                  3825.5536   2489.125      1.537      0.130   -1153.439    8804.546
Mortalidad_Infantil         -18.6792      7.284     -2.565      0.013     -33.249      -4.110
Tasa_Alfabetismo_Femenino   -11.4455     21.377     -0.535      0.594     -54.206      31.315
Tasa_fecundidad             139.0045    279.714      0.497      0.621    -420.507     698.516
==============================================================================
Omnibus:                      103.449   Durbin-Watson:                   2.335
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             2524.104
Skew:                           4.931   Prob(JB):                         0.00
Kurtosis:                      32.142   Cond. No.                     1.28e+03
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.28e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

1. El modelo obtenido es $PIB\_per\_cápita=3825.5536-18.6792* Mortalidad\_Infantil-11.4455*Tasa\_Alfabetismo\_Femenino+139.0045*Tasa\_fecundidad$

2. Análisis estadísticos globales. A partir de los valores de F, R2 y R2 Adj podemos concluir que el modelo es capaz de explicar cerca del 12% de la varianza. F es significativo ($F_{est} < pvalor$). Luego, se rechaza la hipótesis nula de que los coeficientes de regresiones son iguales a cero, por lo que al menos un coeficiente debe ser distinto de cero.

3. Los coeficientes asociados a la variable Mortalidad_Infantil mantienen un test-t estadísticamente significativo ($ t_{est} < pvalor$), por lo que se rechaza la hipótesis nula para estas variables. Para la variable Tasa_Alfabetismo_Femenino y Tasa_fecundidad se acepta la hipótesis nula por lo que su coe
ciente de regresión es igual a cero.
